partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/CompilerType.kt

```diff
@@ -0,0 +1,77 @@
+package org.partiql.planner.internal.typer
+
+import org.partiql.types.PType
+import org.partiql.types.PType.Kind
+
+/**
+ * This is largely just to show that the planner does not need to use [_delegate] ([PType]) directly. Using an
+ * internal representation, we can leverage the APIs of [PType] while carrying some additional information such
+ * as [isMissingValue].
```

Comment:
This is what I was trying to get at with a `PartiQLType (public) -> PType (internal)` thing with a higher class hierarchy, but a fat internal. To be experimented with..